lib/ui/platform_dispatcher.dart

```diff
@@ -1299,6 +1303,102 @@ class PlatformDispatcher {
 
   @Native<Handle Function()>(symbol: 'PlatformConfigurationNativeApi::DefaultRouteName')
   external static String _defaultRouteName();
+
+  /// Scales the `unscaledFontSize`, according to the user's platform
+  /// preferences.
+  ///
+  /// Many platforms allow users to scale text globally for better readability.
+  /// Given the font size the app developer specified in logical pixels, this
+  /// method converts it to the preferred font size (also in logical pixels) that
+  /// accounts for platform-wide text scaling.
+  ///
+  /// Returns null if the platform only supports linear text scaling, or does
+  /// not support platform-wide text scaling. The return value is always
+  /// non-negative and finite if not null.
+  ///
+  /// The scaled value of the same font size input may change if the user changes
+  /// the text scaling preference (in system settings for example). The
+  /// [onTextScaleFactorChanged] callback can be used to monitor such changes.
+  ///
+  /// Instead of directly calling this method, applications should typically use
+  /// [MediaQuery.textScalerOf] to retrive the scaled font size in a widget tree,
+  /// so text in the app resizes properly when the text scaling preference
+  /// changes.
+  double? scaleFontSize(double unscaledFontSize) {
+    assert(unscaledFontSize >= 0);
+    assert(unscaledFontSize.isFinite);
+
+    if (textScaleFactor == 1.0) {
+      return unscaledFontSize;
+    }
+
+    final int unscaledFloor = unscaledFontSize.floor();
```

Comment:
shouldn't the TypeValue.applyDimension be able to handle floating point fontsize? why do we convert to int and do linear intepolation?

Reply:
Yeah it can handle floating point input. But with interpolation we can reduce the cache size, and the Android implementation does the same

Reply:
> Is it really that common to have font sizes in 0.1 increments in your app that we would actually benefit from hitting the cache multiple times?

In an animation that interpolates 2 `TextStyle`s with different font sizes? 

> Also, isn't the point of all of this to not do linear scaling like this does now for the x.0 to x.0 range?

Do you mean [x, x + 1] range where x is an integer? Yeah I've confirmed that the Android implementation does that so I guess we can make that assumption for now?

Reply:
> > Is it really that common to have font sizes in 0.1 increments in your app that we would actually benefit from hitting the cache multiple times?
> 
> In an animation that interpolates 2 `TextStyle`s with different font sizes?
> 
> > Also, isn't the point of all of this to not do linear scaling like this does now for the x.0 to x.0 range?
> 
> Do you mean [x, x + 1] range where x is an integer? Yeah I've confirmed that the Android implementation does that so I guess we can make that assumption for now?

Consider linking to the android implementation in addition to your comment above for why it is safe.

Reply:
The implementation isn't in aosp yet unfortunately: https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/util/TypedValue.java;l=402-420?q=TypedValue.

Reply:
Do we have a policy against linking to code search results? If not then it would help a googler and after the change is merged into aosp someone investigation can probably use the path to find the right result.